functional/FunctionalTest.java

```diff
@@ -211,6 +211,10 @@ public static void writeObject(String urlString, byte[] dataBytes) throws Except
     Request request = requestBuilder
         .url(HttpUrl.parse(urlString))
         .method("PUT", RequestBody.create(null, dataBytes))
+        // Set header 'x-amz-acl' to 'bucket-owner-full-control', so objects created
+        // anonymously, can be downloaded by bucket owner.
+        // Reference: http://prettyplease.me/anonymous-s3-upload-with-full-owner-control/
```

Comment:
Move this comment to above requestBuilder. Also mention this is applicable to aws s3 only.  The reference link is not required.

Reply:
done